tests/servers/test_text_server.h

```diff
@@ -637,6 +637,36 @@ TEST_SUITE("[TextServer]") {
 			}
 		}
 
+		SUBCASE("[TextServer] Unicode letters") {
+			for (int i = 0; i < TextServerManager::get_singleton()->get_interface_count(); i++) {
+				Ref<TextServer> ts = TextServerManager::get_singleton()->get_interface(i);
+				CHECK_FALSE_MESSAGE(ts.is_null(), "Invalid TS interface.");
+
+				static const char32_t *words[19] = { U"-30", U"100", U"10.1", U"10,1", U"1e2", U"1e-2", U"1e2e3", U"0xAB", U"AB", U"Test1", U"1Test", U"Test*1", U"test_testeT", U"test_tes teT", U"\u0643\u0402\u0716\u0b05", U"\u0643\u0402\u0716\u0b05Test\u30aa\u4e21", U"ӒӖӚӜ", U"_test", U"ÂÃÄÅĀĂĄÇĆĈĊ" };
+				static bool words_with_just_letters[19] = { false, false, false, false, false, false, false, false, true, false, false, false, false, false, true, true, true, false, true };
+				for (int j = 0; j < 19; j++) {
```

Comment:
Perhaps it would be more convenient to use a struct and write test cases in a column?

Reply:
Ok, I rewrote the test, please have a look.  
Sadly, I couldn't run the tests locally because of some errors that look like they don't come from my code

<details>

<summary>Backtrace</summary>

```
./godot.linuxbsd.editor.x86_64.llvm --test
[doctest] doctest version is "2.4.11"
[doctest] run with "--help" for options
Could not load project settings.
ERROR: Could not open specified test directory.
   at: make_tests (modules/gdscript/tests/gdscript_test_runner.cpp:326)
===============================================================================
./modules/gdscript/tests/gdscript_test_runner_suite.h:43:
TEST SUITE: [Modules][GDScript]
TEST CASE:  Script compilation and runtime

modules/gdscript/tests/gdscript_test_runner.cpp:191: FATAL ERROR: An error occurred while making the tests.

./modules/gdscript/tests/gdscript_test_runner_suite.h:49: FATAL ERROR: REQUIRE( fail_count == 0 ) is NOT correct!
  values: REQUIRE( -1 == 0 )
  logged: Make sure `*.out` files have expected results.
          All GDScript tests should pass.

ERROR: Could not create directory: res://.godot
   at: make_dir_recursive (core/io/dir_access.cpp:180)
ERROR: Could not create project data directory (res://.godot) in: res://
   at: EditorPaths (editor/editor_paths.cpp:235)
ERROR: Failed to create file res://.godot/.gdignore
   at: EditorPaths (editor/editor_paths.cpp:248)
Could not load project settings.
===============================================================================
./modules/gdscript/tests/test_completion.h:191:
TEST SUITE: [Modules][GDScript][Completion]
TEST CASE:  [Editor] Check suggestion list

./modules/gdscript/tests/test_completion.h:86: FATAL ERROR: Invalid test directory.

ERROR: Can't save resource to empty path. Provide non-empty path or a Resource with non-empty resource_path.
   at: save (core/io/resource_saver.cpp:105)
ERROR: Error saving editor settings to 
   at: save (editor/editor_settings.cpp:1108)
===============================================================================
./modules/gdscript/tests/test_lsp.h:391:
TEST SUITE: [Modules][GDScript][LSP]
TEST CASE:  [workspace][resolve_symbol]

./modules/gdscript/tests/test_lsp.h:90: FATAL ERROR: REQUIRE( err == OK ) is NOT correct!
  values: REQUIRE( 31 == 0 )
  logged: Could not open specified root directory

./modules/gdscript/tests/test_lsp.h:391: FATAL ERROR: test case CRASHED: SIGSEGV - Segmentation violation signal

===============================================================================
[doctest] test cases:  6 |  3 passed | 3 failed | 866 skipped
[doctest] assertions: 53 | 49 passed | 4 failed |
[doctest] Status: FAILURE!

================================================================
handle_crash: Program crashed with signal 11
Engine version: Godot Engine v4.3.dev.custom_build (23191b834e4609baacf19855c3acb6a9f607b30f)
Dumping the backtrace. Please include this when reporting the bug to the project developer.
[1] /lib/x86_64-linux-gnu/libpthread.so.0(+0x14420) [0x7fb31acc7420] (??:0)
[2] ./godot.linuxbsd.editor.x86_64.llvm() [0x2a4144e] (??:0)
[3] ./godot.linuxbsd.editor.x86_64.llvm() [0x2a4380a] (??:0)
[4] ./godot.linuxbsd.editor.x86_64.llvm() [0x2b6e25a] (??:0)
[5] ./godot.linuxbsd.editor.x86_64.llvm() [0x2a4499e] (??:0)
[6] ./godot.linuxbsd.editor.x86_64.llvm() [0x26a1645] (??:0)
[7] ./godot.linuxbsd.editor.x86_64.llvm() [0x261b5b0] (??:0)
[8] /lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xf3) [0x7fb31a990083] (??:0)
[9] ./godot.linuxbsd.editor.x86_64.llvm() [0x261b4ae] (??:0)
-- END OF BACKTRACE --
================================================================
Aborted (core dumped)

```

</details>

Reply:
> Sadly, I couldn't run the tests locally because of some errors that look like they don't come from my code

Test should be run from the repo root not `bin`, some rely on current working directory to load stuff.

Reply:
Argh, I forgot about that. Thanks!  
So, I ran the test and it failed (will look into it)

Reply:
OK, found and fixed the issue in my test